File: OOPs/Inheritance/ModeOfInheritance.java
class Base {
    public int publicVar = 10;
    protected int protectedVar = 20;
    private int privateVar = 30; // Not accessible in Derived
}

class Derived extends Base {
    public void display() {
        System.out.println("Public Var: " + publicVar);
        System.out.println("Protected Var: " + protectedVar);
    }
}

public class ModeOfInheritance {
    public static void main(String[] args) {
        Derived obj = new Derived();
        obj.display();
        System.out.println("Public Var (outside): " + obj.publicVar);
        // System.out.println(obj.protectedVar); // Error: ProtectedVar not accessible
    }
}

File: OOPs/Inheritance/mode_of_inheritance.py
class Base:
    def __init__(self):
        self.publicVar = 10
        self._protectedVar = 20
        self.__privateVar = 30  # Name mangled

class Derived(Base):
    def display(self):
        print("Public Var:", self.publicVar)
        print("Protected Var:", self._protectedVar)
        # print("Private Var:", self.__privateVar)  # Error

# Usage
obj = Derived()
obj.display()
print("Public Var (outside):", obj.publicVar)
# print(obj._protectedVar)  # Accessible but not recommended
